src/app/shared/components/warning/warning.component.ts

```diff
@@ -17,26 +17,32 @@ export class WarningComponent implements OnInit {
   reactorName?: string;
   warningType: Status = Status.outOfRange;
   WarningStyle = WarningStyle;
+
   ngOnInit() {
     console.log(this.safetyStatus);
+    console.log(this.reactorStatus);
     if (!!this.safetyStatus) {
-      this.reactorName = this.safetyStatus.extendedStatus?.name;
-      if (this.safetyStatus.statusCoreTemperature !== Status.inRange) {
-        this.typeOfError = 'Core temperature ';
+      //this.reactorName = this.safetyStatus.extendedStatus?.name;
+      if ( this.safetyStatus.sectionInfo == "Core temperature and production power output has exeeded safe levels") {
+        this.typeOfError = 'Core temperature and production power output ';
         this.warningType = this.safetyStatus.statusCoreTemperature;
-      } else {
-        this.typeOfError = 'Power production output ';
+      } else if (this.safetyStatus.sectionInfo == "Production power output has exeeded safe levels"){
+        this.typeOfError = 'Production power output  ';
         this.warningType = this.safetyStatus.statusPowerProduction;
       }
-    } else if (!!this.reactorStatus) {
-      this.reactorName = this.reactorStatus.name;
-      if (this.reactorStatus?.status.coreTempStatus !== Status.inRange) {
-        this.typeOfError = 'Core temperature ';
-        this.warningType = this.reactorStatus.status.coreTempStatus;
-      } else {
-        this.typeOfError = 'Power production output ';
-        this.warningType = this.reactorStatus.status.powerProdStatus;
+      else if (this.safetyStatus.sectionInfo == "Core temperature has exeeded safe levels"){
+        this.typeOfError = 'Core temperature  ';
+        this.warningType = this.safetyStatus.statusPowerProduction;
```

Comment:
Why using string instead of Status? All you needed to do is : 
- set warning type if power production or temperature has critical status warning type is critical else Status.outofRange
- Add some ngIfs in htm it could look like that 
![image](https://github.com/user-attachments/assets/8dad016c-d4a7-493d-9c4f-996c32da9daf)
if we have safetyStatus we show statusInto as content if we have reactor status we merge text